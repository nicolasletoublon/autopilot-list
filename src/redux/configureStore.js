import thunkMiddleware from 'redux-thunk';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import peopleReducer from './people/reducer';

const middleWares = [thunkMiddleware];

const appReducer = combineReducers({
  people: peopleReducer,
});

const store = createStore(appReducer, applyMiddleware(...middleWares));

export { store };
