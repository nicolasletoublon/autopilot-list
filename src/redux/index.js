/* People REDUCER + ACTIONS */
export { default as list } from './people/reducer';
export { fetchPeopleListThunk } from './people/thunksDispatcher';