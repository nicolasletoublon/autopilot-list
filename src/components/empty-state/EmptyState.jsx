import styled from 'styled-components';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchPeopleListThunk } from '../../redux';

const EmptyStateWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

const Button = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border: 1px solid transparent;
  outline: 0;
  appearance: none;
  transition: all 350ms ease-in;
  padding: 0 16px;
  text-decoration: none;
  border-radius: 4px;
  color: white;
  background-color: #c0c0c0;
  height: 40px;
  cursor: pointer;
  font-size: larger;

  &:hover {
    background-color: #969696;
  }

  &:active {
    background-color: #404040;
  }
`;

Button.displayName = 'LoadingButton';

const EmptyState = () => {
  const { isFetching } = useSelector(({ people }) => people);
  const dispatch = useDispatch();

  const handleLoadDataClick = () => dispatch(fetchPeopleListThunk());

  return (
    <EmptyStateWrapper>
      {isFetching}
      <h4>The list is empty. Click on the button below to load some data.</h4>
      <Button
        onClick={handleLoadDataClick}
        disabled={isFetching}
        id="load-data-button"
      >
        {isFetching ? 'Fetching data...' : 'Load data'}
      </Button>
    </EmptyStateWrapper>
  );
};

export default EmptyState;
