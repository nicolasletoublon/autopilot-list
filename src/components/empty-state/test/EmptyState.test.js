import React from 'react';
import EmptyState from '../EmptyState';
import { shallow, mount } from 'enzyme';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

describe('EmptyState', () => {
  describe('should render his Button with Fetching Data if isFetching is true', () => {
    const state = {
      people: {
        isFetching: true,
      },
    };

    const mockStore = createStore(() => state);

    const comp = (
      <Provider store={mockStore}>
        <EmptyState />;
      </Provider>
    );

    it('snapshot', () => {
      const component = mount(comp);
      expect(component).toMatchSnapshot();
    });

    it('should find the right value in the DOM', () => {
      const component = mount(comp);
      expect(component.find('LoadingButton')).toHaveLength(1);
      expect(component.find('LoadingButton').props().children).toEqual(
        'Fetching data...'
      );
    });
  });

  describe('should render his Button with Load Data if isFetching is false', () => {
    const state = {
      people: {
        isFetching: false,
      },
    };

    const mockStore = createStore(() => state);

    const comp = (
      <Provider store={mockStore}>
        <EmptyState />;
      </Provider>
    );

    it('snapshot', () => {
      const component = mount(comp);
      expect(component).toMatchSnapshot();
    });

    it('should find the right value in the DOM', () => {
      const component = mount(comp);
      expect(component.find('LoadingButton')).toHaveLength(1);
      expect(component.find('LoadingButton').props().children).toEqual(
        'Load data'
      );
    });
  });
});
