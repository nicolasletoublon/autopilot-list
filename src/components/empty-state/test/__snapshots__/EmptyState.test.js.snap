// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render his Button with Fetching Data if isFetching is true snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <EmptyState>
    <styled.div>
      <div
        className="sc-AykKC ePlNtO"
      >
        <h4>
          The list is empty. Click on the button below to load some data.
        </h4>
        <LoadingButton
          disabled={true}
          id="load-data-button"
          onClick={[Function]}
        >
          <button
            className="sc-AykKD hESRCL"
            disabled={true}
            id="load-data-button"
            onClick={[Function]}
          >
            Fetching data...
          </button>
        </LoadingButton>
      </div>
    </styled.div>
  </EmptyState>
  ;
</Provider>
`;

exports[`EmptyState should render his Button with Load Data if isFetching is false snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <EmptyState>
    <styled.div>
      <div
        className="sc-AykKC ePlNtO"
      >
        <h4>
          The list is empty. Click on the button below to load some data.
        </h4>
        <LoadingButton
          disabled={false}
          id="load-data-button"
          onClick={[Function]}
        >
          <button
            className="sc-AykKD hESRCL"
            disabled={false}
            id="load-data-button"
            onClick={[Function]}
          >
            Load data
          </button>
        </LoadingButton>
      </div>
    </styled.div>
  </EmptyState>
  ;
</Provider>
`;
